resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -162,6 +187,8 @@ public DnsNameResolver(
         this.optResourceEnabled = optResourceEnabled;
         this.hostsFileEntriesResolver = checkNotNull(hostsFileEntriesResolver, "hostsFileEntriesResolver");
         this.resolveCache = resolveCache;
+        this.searchDomains = checkNotNull(searchDomains, "searchDomains");
```

Comment:
given that we know it won't be modified, can't we keep it as is and simply add a comment saying that it should be cloned if it's modified at some point ?
or use an immutable arraylist wrapper ?